### PR TITLE
Light CSLA integration

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:

Added stuff from CSLA cdlc to vanilla templates + store + logistics

Affected templates:
vanilla AAF
vanilla civ
vanilla FIA
vanilla Exegermenos rivals
Basically all faction will use some weapons and vehicles from cdlc  

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

Notes:
Didn't add stuff to NATO and CSAT(and LDF) and other rebel and riv templates.
Probably should add stuff to LDF and other rebel/riv templates
Also can call it a continuation of #151

### CODE REVIEW

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_AAF.sqf [hunk 1]
```diff
@@ -119,13 +122,94 @@ if (_hasWs) then {
     _aa pushBack "I_A_Truck_02_aa_lxWS";
 };
 
-["vehiclesLightTanks",  _lightTanks] call _fnc_saveToTemplate;
-["vehiclesCargoTrucks", _cargoTrucks] call _fnc_saveToTemplate;
-["vehiclesAA", _aa] call _fnc_saveToTemplate;
+//If GM cdlc + extra AAF mod
+if (isClass (configFile >> "cfgVehicles" >> "gmx_aaf_m113a2dk_wdl") && _hasGM) then {
```
Comment (Silence-112) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_AAF.sqf:139]: use [hasAddon](https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/blob/stable/A3A/addons/ultimate/functions/Utility/fn_hasAddon.sqf)

```suggestion
private _hasAAFGM = ["gmx_aaf_m113a2dk_wdl"] call A3U_fnc_hasAddon;
if (_hasGM && _hasAAFGM) then {
```

Comment (wersal454) [reply A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_AAF.sqf:139]: Does this make a difference ?

File: A3A/addons/ultimate/config/trader/cfgTraderMods.hpp [hunk 2]
```diff
@@ -75,6 +75,11 @@ class traderMods
         addons[] = {"CUP_Creatures_People_Civil_Russia", "CUP_BaseConfigs", "CUP_AirVehicles_Core"};
         prefix = "cup";
     };
+    class csla : base
+    {
+        addons[] = {"CSLA"};
```
Comment (Silence-112) [inline A3A/addons/ultimate/config/trader/cfgTraderMods.hpp:100]: is the cfgPatches for CSLA literally just "CSLA"?

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_AAF.sqf (hunk 1 above)
Comment (Silence-112) [reply A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_AAF.sqf:139]: makes it more readable, so yes

File: A3A/addons/ultimate/config/trader/cfgTraderMods.hpp (hunk 2 above)
Comment (wersal454) [reply A3A/addons/ultimate/config/trader/cfgTraderMods.hpp:100]: addons,yes, also US stuff uses US85 or something like that

**Comment by wersal454** [comment]
Noticed that there are more markers in blackmarket so I moved vehicles in appropriate categories and fixed _shopJets + added some stuff

**Comment by wersal454** [comment]
Should probably copy changes to other reb/riv templates

**Comment by wersal454** [comment]
https://www.youtube.com/watch?v=X1tyybVuSBU

**Comment by Silence-112** [comment]
There's still a few BIS_AddonInfo.hpp's left

**Pull request merged**